### ISSUE TITLE
Problem with non IDField models after update to 1.6.1

### ISSUE THREAD

**Issue by Chetchaiyan** [octabytes/FireO#168]
Dear All,

I'm facing the following error on model with non IDField 

```
# checking here because `model._id` or `model.id`` are not yet populated
>       if name == self.model._meta.id[0]:
E       TypeError: 'NoneType' object is not subscriptable
```

Please give me some advise on how to fix this.

Note that: same code is working perfectly on v1.5.1

**Comment by AxeemHaider** [comment]
I think this issue is already fixed Can you please give me an example to reproduce this error?

**Comment by Chetchaiyan** [comment]
I already have tables created on previous version (should be v1.5.1). The error appear after I run the upgrade to the newest (v1.6.1) even I don't change any code. 

For more information, My Python version is 3.10

**Comment by dhodun** [comment]
We just came across this as well.

This is a regression from https://github.com/octabytes/FireO/pull/161 and related PRs.

To Repro:

- Use a Non-ID model (i.e. model that uses automatic "id" field.
- Write a filter query against this model _before_ using this model to successfully save or get any object

Will fail:

```python
class TestModel(Model):
    name = TextField()

TestModel.collection.filter("name", "==", "test1").get()
```

Will Succeed (save first)

```python
class TestModel(Model):
    name = TextField()

t = TestModel(name="test")
t.save()

# won't return anything since no match but no error
TestModel.collection.filter("name", "==", "test1").get()
```

Will Succeed (get first)

```python
# insert some data outside FireO so get succeeds
db = firestore.Client()
doc_ref = db.collection(u'test_model').document(u'id1')
doc_ref.set({
  u'name': u'test1'
})

class TestModel(Model):
    name = TextField()

TestModel.collection.get("test_model/id1")

TestModel.collection.filter("name", "==", "test1").get()
```

The issues is that `model._meta.id[0]` is not set for non-ID models until the model is fully initialized in FireO - that is, the Model has been loaded and used to successfully save or query an object.

An early version of my code checked this value for `None` and inferred from that the user had a non-ID model that hadn't been fully initialized. I'll work on a fix PR.

**Impact:**
Any filter queries with `.where()` used on a non-ID model will fail if the the `Model` object is used to query first before saving an object or otherwise fully initializing the Model.

**Comment by dhodun** [comment]
@Chetchaiyan - we are using this fix in 1.6.2 in production. Can you validate it has fixed your issue as well?
